### PR TITLE
Measure GtsfmData I/O time and remove redundant I/O ops

### PR DESCRIPTION
**Pull request by johnwlambert**

### CODE REVIEW

File: gtsfm/scene_optimizer.py [hunk 1]
```diff
@@ -362,55 +364,54 @@ def get_gtsfm_data_with_gt_cameras_and_est_tracks(
...
+        images=images,
+        save_dir=os.path.join(output_dir, "ba_output_gt"),
+    )
 
-    return saving_graph_list
+    end_time = time.time()
+    duration_sec = end_time - start_time
+    logger.info("GtsfmData I/O took %.2f sec.", duration_sec)
+    # Delete old version of React results directory.
+    shutil.rmtree(REACT_RESULTS_PATH)
```
Comment (akshay-krishnan) [inline gtsfm/scene_optimizer.py:412]: why do this delete?

File: gtsfm/scene_optimizer.py [hunk 2]
```diff
@@ -362,55 +364,54 @@ def get_gtsfm_data_with_gt_cameras_and_est_tracks(
...
+    # Save a duplicate copy of the directory in REACT_RESULTS_PATH.
+    shutil.copytree(src=results_path, dst=REACT_RESULTS_PATH)
```
Comment (akshay-krishnan) [inline gtsfm/scene_optimizer.py:414]: could you include this copying in the I/O time above as well?

File: scripts/collect_results.py [hunk 3]
```diff
@@ -136,6 +162,55 @@ def main(user_root: Path, output_fpath: str) -> None:
     save_table_to_tsv(table, headers, output_fpath)
 
 
+def _make_runtime_pie_chart(experiment_roots: Sequence[Path]) -> None:
```
Comment (akshay-krishnan) [inline scripts/collect_results.py:165]: nice!

**Review by akshay-krishnan — APPROVED** [review]

File: gtsfm/scene_optimizer.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py:412]: Turns out copytree only works if the directory doesn't exist, so a delete before is necessary

File: gtsfm/scene_optimizer.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py:414]: Good catch, updated now to include this React I/O time.

**Pull request merged**